recipes/cpython/all/conanfile.py

```diff
@@ -512,6 +512,70 @@ def _msvc_package_copy(self):
         lib_dir_path = os.path.join(self.package_folder, self._msvc_install_subprefix, "Lib").replace("\\", "/")
         self.run(f"{interpreter_path} -c \"import compileall; compileall.compile_dir('{lib_dir_path}')\"")
 
+    def _exact_lib_name(self, folder):
+        possible_extensions = ("a", "so", "dylib", "lib")
+        for file in os.listdir(folder):
+            for extension in possible_extensions:
+                if re.match(f".*\\.{extension}", file):
+                    return file
+        raise ConanException(f"No library files found in {folder}")
+
+    @property
+    def _cmake_module_path(self):
+        if is_msvc(self):
+            # On Windows, `lib` is for Python modules, `libs` is for compiled objects.
+            # Usually CMake modules are packaged with the latter.
+            return os.path.join(self._msvc_install_subprefix, "libs", "cmake")
+        else:
+            return os.path.join("lib", "cmake")
+
+    def _write_cmake_findpython_wrapper_file(self):
+        template = textwrap.dedent("""
+        if (DEFINED Python3_VERSION_STRING)
+            set(_CONAN_PYTHON_SUFFIX "3")
+        else()
+            set(_CONAN_PYTHON_SUFFIX "")
+        endif()
+        set(Python${{_CONAN_PYTHON_SUFFIX}}_EXECUTABLE {})
+        set(Python${{_CONAN_PYTHON_SUFFIX}}_LIBRARY {})
+
+        # Fails if these are set beforehand
+        unset(Python${{_CONAN_PYTHON_SUFFIX}}_INCLUDE_DIRS)
+        unset(Python${{_CONAN_PYTHON_SUFFIX}}_INCLUDE_DIR)
+        
+        include(${{CMAKE_ROOT}}/Modules/FindPython${{_CONAN_PYTHON_SUFFIX}}.cmake)
+                                   
+        # Sanity check: The former comes from FindPython(3), the latter comes from the injected find module
+        if(NOT Python${{_CONAN_PYTHON_SUFFIX}}_VERSION STREQUAL Python${{_CONAN_PYTHON_SUFFIX}}_VERSION_STRING)
+            message(FATAL_ERROR "CMake detected wrong cpython version - this is likely a bug with the cpython Conan package")
+        endif()
+
+        if (TARGET Python${{_CONAN_PYTHON_SUFFIX}}::Module)
+            target_link_libraries(Python${{_CONAN_PYTHON_SUFFIX}}::Module INTERFACE cpython::python)
```

Comment:
This fails for some versions of CMake. Use this pattern instead:
```cmake
set_target_properties(<target> PROPERTIES INTERFACE_LINK_LIBRARIES <libs>)
```

Reply:
Interesting, can do. What version of CMake would that fail for, out of curiosity?

Reply:
Not sure which ones are affected exactly. I've had issues with v3.28.

Reply:
Odd, I thought I was testing with the latest (3.29). I'll test with 3.28 and see what happens.

Reply:
@Ahajha I did not actually encounter any issues with `target_link_libraries` for this recipe. It's more of a best practices / FYI suggestion.

Reply:
Hadn't heard that, I didn't even know there was a different way to link libraries in this way. I'll have to do some reading.

Reply:
Can't find much documentation on it, but it looks like Conan does it internally, so I'll add your suggestion once I test.

Reply:
Done